functions/src/banner/bannerClient.ts

```diff
@@ -0,0 +1,234 @@
+import { Section } from './banner';
+import got, { Response } from 'got';
+import { load } from 'cheerio';
+const ROOT_URL = 'https://banner.uvic.ca';
+const BANNER_SSB_URL = `${ROOT_URL}/StudentRegistrationSsb/ssb`;
+const BANNER_SSB_SEARCH_URL = `${BANNER_SSB_URL}/classSearch/`;
+const BANNER_SSB_SEARCH_GET_SUBJECTS = `${BANNER_SSB_SEARCH_URL}/get_subject`;
+const BANNER_SSD_SEARCH_RESULTS_URL = `${BANNER_SSB_URL}/searchResults/searchResults`;
+const BANNER_SSB_GET_TERMS = `${BANNER_SSB_URL}/classSearch/getTerms`;
+
+type TermsResponse = {
+  code: string;
+  description: string;
+};
+
+export interface BannerResponse<T> {
+  success: boolean;
+  totalCount: number;
+  data: T;
+  pageOffset: number;
+  pageMaxSize: number;
+  sectionsFetchedCount: number;
+  pathMode: string;
+  searchResultsConfigs: SearchResultsConfig[];
+  ztcEncodedImage: null;
+  allowHoldRegistration: null;
+}
+
+export interface SearchResultsConfig {
+  config: string;
+  display: string;
+  title: string;
+  required: boolean;
+  width: string;
+}
+
+export type SearchResultsParams = {
+  /**
+   * e.g. "CSC"
+   */
+  subject?: string;
+  /**
+   * e.g. "201920"
+   */
+  term: string;
+  /**
+   * e.g. "111"
+   */
+  courseNumber?: string;
+  offset?: number;
+  max?: number;
+};
+
+export class BannerClient {
+  cookies: {
+    [key: string]: string[];
+  };
+
+  availableTerms: string[];
+
+  constructor() {
+    this.cookies = {};
+    this.availableTerms = [];
+  }
+
+  getCookie(term: string): string[] | undefined {
+    return this.cookies?.[term];
+  }
+
+  saveCookie(response: Response<string>, term: string): void {
+    // get the cookies
+    const cookies = response.headers['set-cookie'];
+    // if there's no cookies, return
+    if (cookies) {
+      this.cookies = { ...this.cookies, [term]: cookies };
+    }
+  }
+
+  async init() {
```

Comment:
```suggestion
  async init(): void {
```
Prob do this for the rest of the methods too